locator/token_metadata.hh

```diff
@@ -292,13 +297,13 @@ private:
 };
 
 struct topology_change_info {
-    token_metadata target_token_metadata;
-    std::optional<token_metadata> base_token_metadata;
+    lw_shared_ptr<token_metadata> target_token_metadata;
```

Comment:
Can we get back to storing it by value at the end?

Reply:
commented elsewhere